### ISSUE TITLE
Suddenly: Shared shells between different MSYS locations

### ISSUE THREAD

**Issue by LigH-de** [mintty/mintty#598]
I am quite sure that it was possible to run two separate MSYS installations with mintty console each, which used their separate local sh copies with separate mount points. Until yesterday, when I updated some additional software (TortoiseHg, CMake). Suddenly, both mintty consoles launch the same sh shell, according to SysInternals' ProcessExplorer. The second mintty re-uses the sh.exe called by the first mintty. Unfortunately, that causes the same mount points for user directories in both MSYS environments, but I installed both with the purpose to use different home locations and a different set of software. Can you imagine any reason for this "process sharing", and any way to avoid it?

**Comment by mintty** [comment]
I can imagine a number of reason for this but it's hard to analyse remotely. Check your shorcut targets (how do you invoke mintty for either system?), system PATH, whether the two shell instances can "see" each other (`ps -ef`) etc.

In any case the issue is hardly related to mintty, so I'll close it soon, but please report your findings.

**Comment by LigH-de** [comment]
Both instances are called via desktop link, because I can attach a keyboard shortcut.

Link "MSYS - GCC 5.3.0.lnk" executes "H:\MSYS-GCC530\msys.bat --mintty" explicitly in the directory "H:\MSYS-GCC530".
Link "MSYS - GCC 6.2.0.lnk" executes "H:\MSYS-GCC620\msys.bat --mintty" explicitly in the directory "H:\MSYS-GCC620".

According to SysInternals' ProcessExplorer,

a) when I call first "MSYS - GCC 5.3.0.lnk" and then "MSYS - GCC 6.2.0.lnk",

- it calls "H:\MSYS-GCC530\\bin\mintty  /bin/bash -l" in directory "H:\MSYS-GCC530\", which calls "H:\MSYS-GCC530\bin\bash.exe" in directory "H:\MSYS-GCC530\home\_user_\"
- it calls "H:\MSYS-GCC620\\bin\mintty  /bin/bash -l" in directory "H:\MSYS-GCC620\", which calls "H:\MSYS-GCC530\bin\bash.exe" in directory "H:\MSYS-GCC530\home\_user_\"

b) when I call first "MSYS - GCC 6.2.0.lnk" and then "MSYS - GCC 5.3.0.lnk",

- it calls "H:\MSYS-GCC620\\bin\mintty  /bin/bash -l" in directory "H:\MSYS-GCC620\", which calls "H:\MSYS-GCC620\bin\bash.exe" in directory "H:\MSYS-GCC620\home\_user_\"
- it calls "H:\MSYS-GCC530\\bin\mintty  /bin/bash -l" in directory "H:\MSYS-GCC530\", which calls "H:\MSYS-GCC620\bin\bash.exe" in directory "H:\MSYS-GCC620\home\_user_\"

According to "ps -ef", they can see each others' processes, one console as /usr/bin/mintty and the other as /h/MSYS-GCC{530|620}/bin/mintty . Neither directory branch appears in Windows' PATH or in $PATH as printed by printenv.

**Comment by mintty** [comment]
Hmm, as you mention keyboard shortcuts, did you actually invoke the session using them?
If so, try to use the desktop shortcuts instead directly. Any difference?

**Comment by LigH-de** [comment]
No. Assigning a keyboard shortcut to a desktop link is not different to double-clicking it.

**Comment by mintty** [comment]
Was some kind of link between the versions created during upgrade?
Is the respective version you intended to start actually confirmed by `uname -a`?
Speculating further, it could be that some change in 2.6.0 (e.g. there is a workaround for Azure account handling) caused the separation of installations not to work among those particular versions.
As said, just speculating, no further ideas. What about reinstallation?

**Comment by LigH-de** [comment]
Both MSYS environments are prepared "mobile" packages (collected by XhmikosR), you just unzip them into separate directories, and mount points are relative to this location due to missing /etc/fstab files. No active installer application involved.

"Reinstallation" means deleting directory branches and unpacking archives again. So I did. No change.

"uname -a" tells me little. In both cases, the included software is quite similar, differences are less important to me, except the GCC version which differs on purpose.

It is well possible that the mintty version differs between the packages by XhmikosR including GCC 5.x ... 6.1.0 and the one including GCC 6.2.0. So I checked ... no, both identical, v1.1.2. Same for bash, both v3.1.23(1). And the msys.bat files are still identical, too (rev. 2.4).

Now I am even ready to believe that a Microsoft update could have caused changes (e.g. preferring processes or DLLs already in memory)...
